src/Parsers/ExpressionElementParsers.cpp

```diff
@@ -834,10 +828,9 @@ bool ParserNumber::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
 
     auto try_read_float = [&](const char * it, const char * end)
     {
-        char * str_end;
-        errno = 0;    /// Functions strto* don't clear errno.
-        Float64 float_value = std::strtod(it, &str_end);
```

Comment:
It was not guaranteed that the memory range was 0-terminated.
Writing code in C is wrong.